tools/gulp/tasks/screenshots.ts

```diff
@@ -75,49 +96,53 @@ function getLocalScreenshotFiles(dir: string): string[] {
 }
 
 /**
+ * Get processed secure token. The jwt token has 3 parts: header, payload, signature and has format
+ * {jwtHeader}.{jwtPayload}.{jwtSignature}
+ * The three parts is connected by '.', while '.' is not a valid path in firebase database.
+ * Replace all '.' to '/' to make the path valid
+ * Output is {jwtHeader}/{jwtPayload}/{jwtSignature}.
+ * This secure token is used to validate the write access is from our TravisCI under our repo.
+ * All data is written to /$path/$secureToken/$data and after validated the
+ * secure token, the data is moved to /$path/$data in database.
+ */
+function getSecureToken() {
+  return process.env['FIREBASE_ACCESS_TOKEN'].replace(/[.]/g, '/');
+}
+
+/**
  * Upload screenshots to google cloud storage.
  * @param prNumber - The key used in firebase. Here it is the PR number.
- *   If there's no prNumber, we will upload images to 'golds/' folder
- * @param mode - Can be 'test' or 'diff' or 'gold'.
+ * @param mode - Can be 'test' or 'diff' .
  *   If the images are the test results, mode should be 'test'.
  *   If the images are the diff images generated, mode should be 'diff'.
- *   For golds mode should be 'gold'.
  */
-function uploadScreenshots(mode?: 'test' | 'diff' | 'gold', prNumber?: string) {
-  let bucket = openScreenshotsBucket();
-
-  let promises: any[] = [];
+function uploadScreenshotsData(database: firebase.database.Database,
+                               mode: 'test' | 'diff', prNumber: string) {
   let localDir = mode == 'diff' ? path.join(SCREENSHOT_DIR, 'diff') : SCREENSHOT_DIR;
+  let promises: any[] = [];
   getLocalScreenshotFiles(localDir).forEach((file: string) => {
```

Comment:
Same as in the other comment. You could easily `.map` the promises and just return without pushing to an array.